File: src/ts/fs.ts
type DirectoryItemType = string;

export const DIRECTORY_ITEM_TYPE_FILE: DirectoryItemType = "FILE";
export const DIRECTORY_ITEM_TYPE_DIRECTORY_GOBACK: DirectoryItemType = "DIRECTORY_GOBACK";
export const DIRECTORY_ITEM_TYPE_DIRECTORY: DirectoryItemType = "DIRECTORY";
export const DIRECTORY_ITEM_TYPE_UNKNOWN: DirectoryItemType = "UNKNOWN";

interface DirectoryItem {
    name: string;
    itemType: DirectoryItemType;
    size: number;
}

export type {
    DirectoryItemType,
    DirectoryItem,
}

File: src-tauri/src/fs.rs
#[derive(serde::Serialize)]
pub enum DirectoryItemType {
    FILE,
    DIRECTORY,
    UNKNOWN,
}

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DirectoryItem {
    pub name: String,
    pub item_type: DirectoryItemType,
    pub size: u64,
}
